operator/controllers/util.go

```diff
@@ -233,6 +233,9 @@ func createDaemonSet(c client.Client, operation *operatorv1.Operation, namespace
 
 	} else {
 		// Expose /metrics on default (insecure) port
+		if daemonSet.Annotations == nil {
+			daemonSet.Annotations = map[string]string{}
```

Comment:
nit: can't we just do this anyway when initializing the ds first ?

Reply:
Annotations are used only in this branch of code, and I would prefer to keep the branch auto-consistent.
But no problem to change if this is blocking for you